src/compiler/emitter.ts

```diff
@@ -3902,6 +3902,9 @@ namespace ts {
 
         function emitDecorators(parentNode: Node, decorators: NodeArray<Decorator> | undefined) {
             emitList(parentNode, decorators, ListFormat.Decorators);
+            if (decorators && isParameter(parentNode)) {

```

Comment:
When the parent is not a parameter, where does the space come from?

Reply:
It’s notable that this is the _only_ reference to `ListFormat.Decorators` 🤔

Reply:
@andrewbranch I think there is no need to add space for cases like so

```ts
@decorator
class Foo {}
```

```ts
class Foo {
  @decorator
  fn() {}
}
```

Or am I wrong?
Do I need to add logic to writing space for decorators in `emitNodeList`?

Reply:
I’m unsure of the current behavior, but you could easily have

```ts
class Foo {
  @decorator fn() {}
}
```

as an input, and I’m curious what happens with that—something has to write either a newline or a space, and I was wondering if it would make sense to cover the parameter decorator in the same code path. But I’m genuinely unsure where that happens at the moment.

Reply:
The following example has the same error as parameter decorator :(

Before
```ts
class Foo {
  @decorator fn() {}
}
```

After
```ts
class Foo {
  @decoratorfn() {}
}
```

Decorators processed as lists, however, if decorators are located on the same line, there is no option to add space after the last element

https://github.com/microsoft/TypeScript/blob/15c3e99cbd315629e8c2c1444ea879a75aa22d8a/src/compiler/emitter.ts#L4132

only if `format` has `ListFormat.SpaceBetweenBraces` option,
https://github.com/microsoft/TypeScript/blob/15c3e99cbd315629e8c2c1444ea879a75aa22d8a/src/compiler/emitter.ts#L4136
 however, `Decoratos` use only `MultiLine`, `Optional` options. 

What do you think about adding a new option for `Decorators` which will add space after a list?